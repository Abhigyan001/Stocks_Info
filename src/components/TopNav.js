import React from 'react';
import PropTypes from 'prop-types';
import {
  Navbar,
  Nav,
  Form,
} from 'react-bootstrap';

function TopNav({ handleOnSelect }) {
  return (
    <Navbar bg="dark" variant="dark">
      <Navbar.Brand href="/">STOCK INFO</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          <Nav.Link href="/news">Market News</Nav.Link>
        </Nav>
        <Form inline>
          <Form.Control onChange={handleOnSelect} as="select" className="mr-sm-2">
            <option value="active">Most Active Stocks</option>
            <option value="gainers">Most Gaining Stocks </option>
            <option value="losers">Most Losing Stocks </option>
          </Form.Control>
        </Form>
      </Navbar.Collapse>
    </Navbar>
  );
}

TopNav.propTypes = {
  handleOnSelect: PropTypes.func.isRequired,
};

export default TopNav;
