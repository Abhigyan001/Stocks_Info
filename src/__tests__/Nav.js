import React from 'react';
import renderer from 'react-test-renderer';
import TopNav from '../components/Nav';

it('renders TopNav correctly', () => {
  const tree = renderer.create(<TopNav handleOnSelect={jest.fn()} />);
  expect(tree).toMatchSnapshot();
});
