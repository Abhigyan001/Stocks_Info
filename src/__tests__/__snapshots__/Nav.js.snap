// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TopNav correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    STOCKS INFO
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="/news"
        href="/news"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Market News
      </a>
    </div>
    <form
      className="form-inline"
    >
      <select
        className="mr-sm-2 form-control"
        onChange={[MockFunction]}
      >
        <option
          value="active"
        >
          Most Active Stocks
        </option>
        <option
          value="gainers"
        >
          Most Gaining Stocks 
        </option>
        <option
          value="losers"
        >
          Most Losing Stocks 
        </option>
      </select>
    </form>
  </div>
</nav>
`;
